kubespawner/spawner.py

```diff
@@ -1773,9 +1892,12 @@ def _start_watching_events(self, replace=False):
         If replace=True, a running pod reflector will be stopped
         and a new one started (for recovering from possible errors).
         """
+        event_reflector_class = EventReflector
+        if self.enable_user_namespaces:
+            event_reflector_class = MultiNamespaceEventReflector
```

Comment:
Depending on the resolution of the above discussion, this may be a single class adding `kwargs["omit_namespace"] = True`

Reply:
Done!